### PR TITLE
CURA-7090/Added function reEvaluateDismissedPackages() in the Package Manager

### PR DESCRIPTION
**Pull request by DonExo**
Added two functions to the Package Manager:
1. A function that marks a package as "dismissed" and puts it in a list called `dismissed` in the `packages.json` file in the user's local data folder.
2. A function that re-evaluates the dismissed incompatible packages, and if needed removes them from the dismissed_packages list.

Is being used in: https://github.com/Ultimaker/Cura/pull/7011

### CODE REVIEW

File: UM/PackageManager.py [hunk 1]
```diff
@@ -345,9 +345,26 @@ def getAllInstalledPackagesInfo(self) -> Dict[str, List[Dict[str, Any]]]:
...
+
     def getDismissedPackages(self) -> List[str]:
         return list(self._dismissed_packages)
 
+    def removeFromDismissedPackages(self, package: str) -> None:
+        if package in self._dismissed_packages:
+            self._dismissed_packages.remove(package)
+            Logger.debug("Removed package [%s] from the dismissed packages list" % package)
+
+    def reEvaluateDismissedPackages(self, data) -> None:
```
Comment (konskarm) [inline UM/PackageManager.py]: Add type of "data"

File: UM/PackageManager.py [hunk 2]
```diff
@@ -345,9 +345,26 @@ def getAllInstalledPackagesInfo(self) -> Dict[str, List[Dict[str, Any]]]:
...
+        dismissed_packages = self.getDismissedPackages()
+        for dismissed in dismissed_packages:
+            for item in data:
+                if item['package_id'] == dismissed:
```
Comment (konskarm) [inline UM/PackageManager.py]:
```suggestion
                if item["package_id"] == dismissed:
```

File: UM/PackageManager.py [hunk 3]
```diff
@@ -11,6 +11,9 @@
 
 from PyQt5.QtCore import pyqtSlot, QObject, pyqtSignal, QUrl, pyqtProperty
 
+from cura import ApplicationMetadata
```
Comment (ninovanhooff) [inline UM/PackageManager.py]: Cura shouldn't be used from Uranium, right? Are you still working on this?

Comment (DonExo) [reply UM/PackageManager.py]: That was a mistake, it has been deleted.
The `sdk_version` is now passed from the calling cura function.

**Pull request merged**